WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -140,7 +141,23 @@ private void StepConnectionConfirmationPhase()
 					}
 					else
 					{
-						round.SetPhase(Phase.OutputRegistration);
+						await foreach (var offendingAlices in CheckTxoSpendStatusAsync(round, cancel).ConfigureAwait(false))
+						{
+							if (offendingAlices.Any())
+							{
+								var removed = round.Alices.RemoveAll(x => offendingAlices.Contains(x));
+								round.LogInfo($"There were {removed} alices removed because they spent the registered UTXO.");
+							}
+						}
+						if (round.InputCount < Config.MinInputCountByRound)
+						{
+							round.SetPhase(Phase.Ended);
+							round.LogInfo($"Not enough inputs ({round.InputCount}) in {nameof(Phase.ConnectionConfirmation)} phase.");
+						}
```

Comment:
I think this would have simpler control flow if this was not duplicated with the branch at line 137, but instead the `else` at line 141 was just an `(round.InputCount >= Config.MinInputCountByRound)` that first removes inputs, and then just goes into the original if/else.

then the flow would be: if the timeout is reached, first remove any non-confirmed inputs, then check confirmed inputs for double spend and remove those, then check if there are enough inputs left, and unconditionally change the phase either to output registration or ended accordingly

Reply:
Right. Make sense.

Reply:
Done.